### PR TITLE
Alternative react-redux structure with tests

### PR DESCRIPTION
**Pull request by jrd404**
@gbzaid2 @tonyguo665 

**For immediate review (it's a lot):**

- Added alternative redux structure to @tonyguo665 
- Example components configured with `react-router` routing
- Example tests mocking redux store/reducers
- Fully Functional NavBar component (sans SearchBar)

### CODE REVIEW

File: src/components/__tests__/ExampleForm.test.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+describe("ExampleForm", () => {
+  const mockStore = createStore(rootReducer, { data: [] });
+  const getWrapper = () =>
+    mount(
+      <Provider store={mockStore}>
+        <ExampleForm />
+      </Provider>
+    );
+
+  it("should be true", () => {});
```
Comment (tonyguo665) [inline src/components/__tests__/ExampleForm.test.jsx]: What is this empty test?

File: src/components/ExampleForm.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { connect } from "react-redux";
+import { addItem } from "../js/actions/index";
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    addItem: (data) => dispatch(addItem(data))
+  };
+};
+
+class ConnectedExampleForm extends Component {
```
Comment (tonyguo665) [inline src/components/ExampleForm.jsx]: Can we switch this to a functional component and use hooks for consistency of structure?

File: src/components/__tests__/ExampleForm.test.jsx (hunk 1 above)
Comment (tonyguo665) [reply src/components/__tests__/ExampleForm.test.jsx]: @jarrodchung-dev

File: src/components/ExampleForm.jsx (hunk 2 above)
Comment (tonyguo665) [reply src/components/ExampleForm.jsx]: @jarrodchung-dev

**Review by jrd404 — COMMENTED** [review]
Viewed all changes (no actual code conflicts, just file deletions)

**Review by tonyguo665 — APPROVED** [review]

**Pull request closed without merging**